src/content/post/20161129_git_reflog_amend.md

```diff
@@ -11,104 +11,103 @@ author      = "Marcello"
 But the `--amend` option can also be used to include *new code* in the last commit.
 
-For example.
+For example:
 Let's say you've just created a commit but you've also forgot to add a block comment to that very method you've written, and you don't want to have another commit in the history, for various reasons. You have many choices at this point, including creating a new commit with the new lines and squash it with the previous one using the `git rebase` command.
 
 But you can also do this:
 
-<pre>
+``` bash
 # Add new line and prepare them for the commit.
 git add -p
 # Amend the previous commit and add to it the staged modifications.
 git commit --amend
-</pre>
+```
 
 A this point, you are asked to edit the previous commit message and the staged edits will be included in it. Useful, isnt'it?
 
 ## The side effect
 
-This little trick can be very handy in many situations, but it can also be a very sharp double-edge blade because, as you probably have guessed, you can use the amend on *any* commit including, for example, a *merge commit*.
+This little trick can be very handy in many situations, but it can also be a very sharp double-edged blade because, as you probably have guessed, you can use the amend on *any* commit including, for example, a *merge commit*.
 
 So, here's a possible sequence of commands that could generate a bad situation.
 
-<pre>
+``` bash
 # You add some modifications.
 git add -p
 # Then create a new commit
 git commit -m 'Added new public method'
 # Then merge the last develop in your working branch
 git fetch --all
 git merge origin/develop
-</pre>
+```
 
 So far the history of your branch could be something like:
 
-<pre>
+``` bash
 ~$ git log --oneline
 
 b1748d6 Merge remote-tracking branch 'origin/develop' into feature/my_branch
 a69fa15 Added new public method
-</pre>
+```
 
 Now you accidendally amend the next modifications instead of creating a new commit.
 
-<pre>
+``` bash
 # Write a block comment for your new method and stage the new modifcations.
 git add -p
 # But instead of creating a new commit, you use the --amend option.
 git commit --amend
-</pre>
+```
 
-At this point, the history will be almost the same, but the hash of the *merge commit* will be different and that commmit will includeyour new block comment too.
+At this point, the history will be almost the same, but the hash of the *merge commit* will be different and that commmit will include your new block comment too.
 
-<pre>
+``` bash
 ~$ git log --oneline
 
 f5a5377 Merge remote-tracking branch 'origin/develop' into feature/my_branch
 a69fa15 Added new public method
-</pre>
+```
 
 That's bad.
 
 ## The git reflog solution
 
-Because you want to restore the situation before the amend  you can decide to use the `git reflog` in combination with a `git reset`.
+Because you want to restore the situation, before the amend  you can decide to use the `git reflog` in combination with a `git reset`.
```

Comment:
Sorry for not taking this note before: "because" is always in response to "why". The word you are searching is "Since".

`Since you want to ...`